### PR TITLE
[Core] Refactor OLED to allow easy addition of other types 

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Change from `OLED_DRIVER_ENABLE` to just `OLED_ENABLE`.
Then specify the specific OLED driver with `OLED_DRIVER = SSD1306`.
Currently it defaults to SSD1306 if `OLED_DRIVER` is not specified.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Refactor OLED to allow easy addition of other types 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: common_features.mk [hunk 1]
```diff
@@ -600,11 +600,26 @@ ifeq ($(strip $(HD44780_ENABLE)), yes)
...
+        OPT_DEFS += -DOLED_ENABLE
+        COMMON_VPATH += $(DRIVER_PATH)/oled
+
+        ifeq ($(strip $(OLED_DRIVER)), SSD1306)
+            SRC += ssd1306_sh1106.c
+        endif
+        # add extra deps
+        ifeq ($(strip $(OLED_DRIVER)), SSD1306)
+            QUANTUM_LIB_SRC += i2c_master.c
+        endif
```
Comment (drashna) [inline common_features.mk:618]:
```suggestion
        OPT_DEFS += -DOLED_DRIVER_$(strip $(shell echo $(OLED_DRIVER) | tr '[:lower:]' '[:upper:]'))

        ifeq ($(strip $(OLED_DRIVER)), SSD1306)
            SRC += ssd1306_sh1106.c
            QUANTUM_LIB_SRC += i2c_master.c
        endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: common_features.mk (hunk 1 above)
Comment (Xelus22) [reply common_features.mk:618]: Would we prefer this as a `#define` or a `rules.mk` addition? 
I was just basing this off of how WS2812 driver is specified.

Comment (drashna) [reply common_features.mk:619]: This handles it closer to how rgb matrix handles the drivers, which for this, is better, I think.

**Comment by drashna** [comment]
This has a bunch of merge conflicts that need resolution.

**Comment by drashna** [comment]
And tested, doesn't seem to effect existing functionality, so 👍🏻

**Comment by Xelus22** [comment]
From a `qmk multibuild` reveals that this breaks helix due to it already using `OLED_ENABLE` in the local_features.mk

**Comment by drashna** [comment]
@mtei @MakotoKurauchi tagging you since, helix is impacted by this, negatively.

**Comment by mtei** [comment]
@Xelus22 , please merge https://github.com/mtei/qmk_firmware/tree/fix_helix_in_13454_oled_enale

**Comment by Xelus22** [comment]
> 
> 
> @Xelus22 , please merge [https://github.com/mtei/qmk_firmware/tree/fix_helix_in_13454_oled_enale](https://github.com/mtei/qmk_firmware/tree/fix_helix_in_13454_oled_enale?rgh-link-date=2021-08-18T11%3A44%3A09Z)

@mtei done 👍

**Review by mtei — APPROVED** [review]
Weak approve.

**Review by tzarc — APPROVED** [review]

**Comment by Conor-Burns** [comment]
Hi it seems like this missed the bongocat keymap on the 0xCB Static https://github.com/qmk/qmk_firmware/pull/15232

**Pull request merged**